src/actions/reviewsActions.ts

```diff
@@ -0,0 +1,88 @@
+import { createAction } from '@reduxjs/toolkit';
+import { Dispatch } from 'redux';
+import reviewsService, { Query } from '../services/reviewsService';
+import { ReviewStats } from '../typings/reviewStats';
+import { ReviewsComments } from '../typings/reviewsComments';
+import { initialStateType } from '../reducers/reviewsReducer';
+
+export const requestReviewStats = createAction('REVIEWS/REQUEST_STATS');
+export const receiveReviewStats = createAction<ReviewStats>('REVIEWS/REQUEST_STATS_SUCCESS');
+export const fetchReviewStatsError = createAction('REVIEWS/REQUEST_STATS_ERROR');
+
+export function fetchReviewStats() {
+  return async (dispatch: Dispatch) => {
+    dispatch(requestReviewStats());
+
+    try {
+      const reviewStats = await reviewsService.getReviewStats();
+      dispatch(receiveReviewStats(reviewStats))
+    } catch (e) {
+      dispatch(fetchReviewStatsError(e))
+    }
+
+  }
+}
+
+export const requestReviewComments = createAction('REVIEWS/REQUEST_COMMENTS');
+export const receiveReviewComments = createAction<ReviewsComments>('REVIEWS/REQUEST_COMMENTS_SUCCESS');
+export const fetchReviewCommentsError = createAction('REVIEWS/REQUEST_COMMENTS_ERROR');
+
+export function fetchReviewComments(page = 1, limit = 20): any {
+  return async (dispatch: Dispatch) => {
+    dispatch(requestReviewComments());
+
+    try {
+      const reviewComments = await reviewsService.getReviewComments({ page, limit });
+      dispatch(receiveReviewComments(reviewComments))
+    } catch (e) {
+      dispatch(fetchReviewCommentsError(e))
+    }
+
+  }
+}
+
+export const requestFilterAndSortReviewComments = createAction('REVIEWS/REQUEST_FILTER_AND_SORT');
+export const receiveFilterAndSortReviewComments = createAction<{
+  reviewComments: ReviewsComments,
```

Comment:
try to avoid inline typing and create an interfaces for it

Reply:
nice one!